src/onnx/parse_gridsample.cpp

```diff
@@ -0,0 +1,432 @@
+                   const instruction_ref& grid,
+                   bool align,
+                   std::string&& padding)
+        : grid_sampler(input, grid, align, std::move(padding))
+    {
+    }
+
+    void setup(const onnx_parser::node_info& info) override
+    {
+        grid_sampler::setup(info);
+        m_floor_x              = info.add_common_op("floor", m_unnorm_x);
+        m_floor_y              = info.add_common_op("floor", m_unnorm_y);
+        m_ceil_x               = info.add_common_op("add", m_floor_x, m_one_l);
+        m_ceil_y               = info.add_common_op("add", m_floor_y, m_one_l);
+        auto fract_x           = info.add_common_op("sub", m_unnorm_x, m_floor_x);
+        auto fract_y           = info.add_common_op("sub", m_unnorm_y, m_floor_y);
+        auto one_minus_fract_x = info.add_common_op("sub", m_one_l, fract_x);
+        auto one_minus_fract_y = info.add_common_op("sub", m_one_l, fract_y);
+        m_corner_weights[0]    = info.add_common_op("mul", one_minus_fract_y, one_minus_fract_x);
+        m_corner_weights[1]    = info.add_common_op("mul", one_minus_fract_y, fract_x);
+        m_corner_weights[2]    = info.add_common_op("mul", fract_y, one_minus_fract_x);
+        m_corner_weights[3]    = info.add_common_op("mul", fract_y, fract_x);
+    }
+
+    instruction_ref sample(const onnx_parser::node_info& info) override
+    {
+        setup(info);
+        std::array<std::vector<instruction_ref>, 4> indices_all;
+        std::array<std::vector<instruction_ref>, 4> validation_all;
+        std::vector<instruction_ref> weight_indices;
+
+        static auto nhw_shape = migraphx::shape{migraphx::shape::int64_type, {3}};
+        for(size_t n = 0; n < m_batch; n++)
+        {
+            for(size_t h = 0; h < m_out_height; h++)
+            {
+                for(size_t w = 0; w < m_out_width; w++)
+                {
+                    auto nhw = info.add_literal(migraphx::literal{nhw_shape, {n, h, w}});
+                    auto y0  = info.add_instruction(make_op("gathernd"), m_floor_y, nhw);
+                    auto x0  = info.add_instruction(make_op("gathernd"), m_floor_x, nhw);
+                    auto y1  = info.add_instruction(make_op("gathernd"), m_ceil_y, nhw);
+                    auto x1  = info.add_instruction(make_op("gathernd"), m_ceil_x, nhw);
+                    weight_indices.push_back(nhw);
+                    for(size_t c = 0; c < m_channel; c++)
+                    {
+                        update_indices(info, y0, x0, n, c, indices_all.at(0), validation_all.at(0));
+                        update_indices(info, y0, x1, n, c, indices_all.at(1), validation_all.at(1));
+                        update_indices(info, y1, x0, n, c, indices_all.at(2), validation_all.at(2));
+                        update_indices(info, y1, x1, n, c, indices_all.at(3), validation_all.at(3));
+                    }
+                }
+            }
+        }
```

Comment:
Same comment as above to avoid O(n^4). There seems like we're only doing ops after finding batch, height, width.